kratos/geometries/triangle_3d_3.h

```diff
@@ -879,7 +875,66 @@ template<class TPointType> class Triangle3D3
         const CoordinatesArrayType& rPoint
         ) const override
     {
-        return PointLocalCoordinatesImplementation(rResult, rPoint);
+        // Initialize
+        rResult = ZeroVector(3);
+
+        // Tangent vectors
+        const array_1d<double, 3> tangent_xi  = this->GetPoint(1) - this->GetPoint(0);
+        const array_1d<double, 3> tangent_eta = this->GetPoint(2) - this->GetPoint(0);
+
+        // We compute the normal to check the normal distances between the point and the triangles, so we can discard that is on the triangle
+        array_1d<double, 3> normal;
+        MathUtils<double>::UnitCrossProduct(normal, tangent_xi, tangent_eta);
+
+        // The center of the geometry
+        const auto& r_center = this->Center();
+
+        // We compute the distance, if it is not in the pane we
+        CoordinatesArrayType point_projected = rPoint;
+        const array_1d<double,3> vector_points = rPoint - r_center.Coordinates();
+        const double distance = inner_prod(vector_points, normal);
+        if (std::abs(distance) > std::numeric_limits<double>::epsilon()) { // Not in the plane, projecting

```

Comment:
wait, doesn't this function assume that the point is in-plane?

I think this is a behavior-change?